packages/dht/src/connection/websocket/WebsocketConnector.ts

```diff
@@ -143,10 +143,10 @@ export class WebsocketConnector {
     public async start(): Promise<void> {
         if (!this.abortController.signal.aborted && this.websocketServer) {
             this.websocketServer.on('connected', (connection: IConnection) => {
-                const serverSocket = connection as unknown as ServerWebsocket
-                const query = serverSocket.resourceURL.query as unknown as (ParsedUrlQuery | null)
+                const serverSocket = connection as unknown as WebsocketServerConnection
+                const query = queryString.parse(serverSocket.resourceURL.query as string)
```

Comment:
If there is on query (the `?foo=bar` part), will this `value` be empty or `null` or `undefined`? If can be `null` or `undefined`, could adjust the type. If not, could remove the question mark at line `148`.

Reply:
The `serverSocket.resourceURL.query` is be `null` if there is no query string. E.g. if we do `websocat 'ws://10.200.10.1:40500/'`.

Reply:
The `serverSocket.resourceURL.query` is be `null` if there is no query string. E.g. if we do `websocat 'ws://10.200.10.1:40500/'`.

Reply:
const query = {} at line 147 in this case

Reply:
This would maybe work, and be also type-correct:

```suggestion
                const query = queryString.parse(serverSocket.resourceURL.query ?? '')
```